module/condition/src/Infrastructure/Persistence/Query/DbalConditionSetQuery.php

```diff
@@ -51,16 +51,14 @@ public function getDataSet(Language $language): DbalDataSet
     /**
      * @return array
      */
-    public function findNumericConditionRelations(AttributeId $attributeId): array
+    public function findAttributeIdConditionRelations(AttributeId $attributeId): array
     {
         $qb = $this->connection->createQueryBuilder();
 
         $records = $qb->select('id')
             ->from(self::TABLE)
-            ->where($qb->expr()->eq('conditions->0->>\'type\'', ':type'))
-            ->where($qb->expr()->eq('conditions->0->>\'attribute\'', ':attribute_id'))
-            ->setParameter(':type', 'NUMERIC_ATTRIBUTE_VALUE_CONDITION')
-            ->setParameter(':attribute_id', $attributeId->getValue())
+            ->where('conditions::text ILIKE :attribute_id')
```

Comment:
I don't really like the solution. Though unlikely it may occur that there will be text or id assigned to another type of entity matching.
What is the reasoning for that?